content/blog/2021-02-18-dvc-2-0-pre-release.md

```diff
@@ -0,0 +1,282 @@
+
+descriptionLong: |
+  The new release is a result of our learnings from our users. There are four
+  major features are coming:
+
+  🔗 ML pipelines templating and iterative foreach stages
+
+  🧪 Lightweight ML experiments
+
+  📍 ML model checkpoints
+
+  📈 Dvc-live - metrics logging
+
+picture: 2021-02-18/dvc-2-0-pre-release.png
+pictureComment: DVC 2.0 Pre-Release
+author: dmitry_petrov
+commentsUrl: https://discuss.dvc.org/t/dvc-3-years-anniversary-and-1-0-pre-release/374
+tags:
+  - Release
+  - MLOps
+  - DataOps
+---
+
+## Install
+
+First things first. You can install the 1.0 pre-release from the master branch
+in our repo (instruction [here](https://dvc.org/doc/install/pre-release)) or
+through pip:
+
+```bash
+pip install --upgrade --pre dvc
+```
+
+## ML pipelines parametrization and foreach stages
+
+After introducing the multi-stage pipeline file `dvc.yaml`, it was quickly
+adopted among DVC users. The DVC team got tons of positive feedback from users
+as well as feature requests.
+
+### Parameters
+
+The most requested feature was the ability to use parameters in the pipeline
+file. So, you can pass the same seed value or filename to multiple stages in the
+pipeline.
+
+```yaml
+var:
+    train_matrix: train.pkl
+    test_matrix: test.pkl
+    seed: 20210215
+
+...
+
+stages:
+    process:
+        cmd: python process.py \
+                --seed ${seed} \
+                --train ${train_matrix} \
+                --test ${test_matrix}
+        outs:
+        - ${test_matrix}
+        - ${train_matrix}
+
+        ...
+
+    train:
+        cmd: python train.py ${train_matrix} --seed ${seed}
+        deps:
+        - ${train_matrix}
+```
+
+Also, it gives an ability to localize all important parameters in a single `var`
+block and play with them. This is a natural thing to do for scenarios like NLP
+or when hyperparameter optimization is happening not only in the model training
+code but in the data processing as well.
+
+### Pipeline params from params.yaml
+
+It is quite common to define pipeline parameters in a config file or the
+parameters file `params.yaml` instead of in the pipeline file `dvc.yaml`.
+Parameters defined in `params.yaml` can also be used in `dvc.yaml`.
+
+Params.yaml:
+
+```yaml
+models:
+  us:
+    thresh: 10
+    filename: 'model-us.hdf5'
+```
+
+Pipeline file `dvc.yaml`:
+
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python script.py
+        --out ${models.us.filename}
+        --thresh ${models.us.thresh}
+    outs:
+      - ${models.us.filename}
+```
+
+DVC properly tracks params dependencies for each stage starting from the
+previous DVC version 1.0. See
+[params option](https://dvc.org/doc/command-reference/run#for-displaying-and-comparing-data-science-experiments)
+in `dvc run` for more details.
+
+### Iterating over params with foreach stages
+
+Iterating over params was a frequently requested feature. Now users can define
+multiple similar stages by a single command.
```

Comment:
Maybe
```suggestion
multiple similar stages with a templetized command.
```
Or "parameterized"

Reply:
Should it be "templatized" instead of "templetized" (not sure either is a real world 🤷 )?

Reply:
Good point, with `a` it has SOME results out there: https://trends.google.com/trends/explore?geo=US&q=templatize,templetize

Or maybe we should say "variable command"?